### PR TITLE
v.generalize: Add tests

### PR DESCRIPTION
**Pull request by Kiraschild**
This PR adds test cases for the v.generalize module, including Hermite interpolation and Chaiken's algorithm tests. The tests verify that the output lines meet the expected properties.

### CODE REVIEW

**Review by petrasovaa — CHANGES_REQUESTED** [review]
You can leave the headers out (-n flag), they are not needed, just cluttering the tests. Also include the reference values in the test file directly, no need for more files.

File: vector/v.generalize/testsuite/v_generalize_test.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+        info_output = grass.read_command(
+            "v.out.ascii", input=f"{vector_name}_vertices", format="point"
+        )
+        for line in info_output.splitlines():
+            if line.strip():
+                x, y, _ = line.split()
+                vertices.append((float(x), float(y)))
+        return vertices
+
+    def count_vertices(self, vector_name):
```
Comment (petrasovaa) [inline vector/v.generalize/testsuite/v_generalize_test.py:97]: You can use gs.vector_info_topo instead

File: vector/v.generalize/testsuite/v_generalize_test.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+        )
+        self.assertVectorExists(f"{vector_name}_vertices")
+
+        # Get the number of points (vertices) using v.info
+        info_output = grass.read_command(
+            "v.info", map=f"{vector_name}_vertices", flags="t"
+        )
+        # Extract the number of points from the output
+        num_vertices = 0
+        print(info_output)
```
Comment (petrasovaa) [inline vector/v.generalize/testsuite/v_generalize_test.py:116]: No prints in the tests

Comment (Kiraschild) [reply vector/v.generalize/testsuite/v_generalize_test.py]: I have made the required changes in the test cases.
1. I have removed the print statements
2. Used the -n flag in v.in.ascii to ignore the headers
3. Removed the coordinate files and included the reference values in the tests itself.
4. Instead of parsing the file and counting the vertices. I have used gs.vector_info_topo for the same.

**Comment by petrasovaa** [comment]
There is additional test of v.cluster in this PR, please remove that from here.

**Comment by Kiraschild** [comment]
> There is additional test of v.cluster in this PR, please remove that from here.

I have done the required changes. Pls review it and do let me know about the additional changes.

**Comment by petrasovaa** [comment]
> > There is additional test of v.cluster in this PR, please remove that from here.
> 
> I have done the required changes. Pls review it and do let me know about the additional changes.

Do you plan to test other algorithms in there?

**Comment by Kiraschild** [comment]
> > > There is additional test of v.cluster in this PR, please remove that from here.
> > 
> > 
> > I have done the required changes. Pls review it and do let me know about the additional changes.
> 
> Do you plan to test other algorithms in there?

I’d be happy to expand test coverage for additional algorithms in v.generalize. Could you clarify which specific algorithms or edge cases you’d like prioritized? Alternatively, if the current tests are sufficient for merging, I can address the remaining issues in a follow-up pull request.

**Review by petrasovaa — APPROVED** [review]
I will merge this for now, more tests can be added later.
